homeassistant/components/uptime_kuma/config_flow.py

```diff
@@ -124,3 +124,42 @@ async def async_step_reauth_confirm(
             ),
             errors=errors,
         )
+
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Confirm reauthentication dialog."""
+        errors: dict[str, str] = {}
+
+        entry = self._get_reconfigure_entry()
+
+        if user_input is not None:
+            url = URL(user_input[CONF_URL])
+            self._async_abort_entries_match({CONF_URL: url.human_repr()})
+
+            session = async_get_clientsession(self.hass, user_input[CONF_VERIFY_SSL])
+            uptime_kuma = UptimeKuma(session, url, user_input[CONF_API_KEY])
+
+            try:
+                await uptime_kuma.metrics()
+            except UptimeKumaAuthenticationException:
+                errors["base"] = "invalid_auth"
+            except UptimeKumaException:
+                errors["base"] = "cannot_connect"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
```

Comment:
Maybe break this out to it's own function so it can be shared between the different steps which seems to use the same e.g. make the code a bit more dry.